### PR TITLE
Refactor report errors for the APIv2 (closes #374)

### LINKED ISSUE #374: Refactor report errors for the APIv2
- [x] Move report-related errors code to `errors/test-run` and the runtime errors code to `errors/runtime`
- [x] Prefix all legacy API errors and categories with `legacy`, move them to `legacy/reporter/errors`
- [x] Use`LegacyTestRunError` type for legacy API errors and delegate processing to the legacy templates
- [x] Introduce `TestRunErrorFormattableAdapter`
- [x] Redesign error decorator so it fits need of both API versions
- [x] Check all reporters and introduce temporary testing infrastructure
  - [x] spec
  - [x] list
  - [x] minimal
  - [x] json
  - [x] xunit
- [ ] Implement generic errors
  - [x] External assertion lib errors
  - [x] Test code error
  - [x] Uncaught error on page
  - [ ] Callsites for assertion errors (filter `node_modules` and internal callsites)
  - [x] Page load error (see **Handling page error** below)

- [x] Add decorator for error subtitles (e.g. for beforeEach/afterEach headers)
- [x] Wrap null, undefined number, etc. with `<code>` in messages

## Handling page error
-  Send `about:page-error` and add pending error (this is blocking on DevExpress/testcafe-hammerhead#339)
- Then if error was not handled use standard `done` mechanism

### PR DESCRIPTION
**Pull request by AlexanderMoskovkin**
* Callsites for assertion errors (filter node_modules and internal callsites)

/cc @inikulin @VasilyStrelyaev

### CODE REVIEW

File: test/functional/fixtures/api/es-next/generic-errors/testcafe-fixtures/external-assertion-lib-errors-test.js [hunk 1]
```diff
@@ -1,7 +1,11 @@
 import assert from 'assert';
-import { expect } from 'chai';
+import { expect, config as chaiConfig } from 'chai';
 import { assertionError } from './helpers';
 
+// NOTE: set this flag to check that TestCafe adds a correct callsite to the
+// report when there are chai module files in the top of the error stack.
```
Comment (VasilyStrelyaev) [inline test/functional/fixtures/api/es-next/generic-errors/testcafe-fixtures/external-assertion-lib-errors-test.js]:
```
// report when there are chai module files on top of the error stack.
```

**Comment by VasilyStrelyaev** [comment]
\r-

**Comment by VasilyStrelyaev** [comment]
lgtm

File: src/errors/process-test-fn-error.js [hunk 2]
```diff
@@ -8,6 +10,13 @@ import {
 } from './test-run';
 
 
+function isAssertionErrorCallsiteFrame (frame) {
+    var filename = frame.getFileName();
+
+    return !new RegExp(`${escapeRe(sep)}node_modules${escapeRe(sep)}`).test(filename);
```
Comment (inikulin) [inline src/errors/process-test-fn-error.js]: We need to filter node internal modules as well see: https://github.com/DevExpress/testcafe/blob/master/src/errors/stack-filter.js#L23 and https://github.com/DevExpress/testcafe/blob/master/src/errors/stack-filter.js#L24

**Comment by inikulin** [comment]
\r-

**Comment by AlexanderMoskovkin** [comment]
FPR
/cc @inikulin @VasilyStrelyaev

**Comment by VasilyStrelyaev** [comment]
lgtm

File: src/errors/process-test-fn-error.js [hunk 3]
```diff
@@ -8,6 +9,19 @@ import {
 } from './test-run';
 
 
+const INTERNAL = 'internal/';
+
+function isAssertionErrorCallsiteFrame (frame) {
+    var filename = frame.getFileName();
+
+    // NOTE: filter node and assertion libraries internals
```
Comment (VasilyStrelyaev) [inline src/errors/process-test-fn-error.js]:
```
    // NOTE: filter out the internals of node.js and assertion libraries
```

**Comment by VasilyStrelyaev** [comment]
\r-

**Comment by AlexanderMoskovkin** [comment]
FPR

**Comment by AlexanderMoskovkin** [comment]
FPR

File: src/errors/test-run/templates.js [hunk 4]
```diff
@@ -223,7 +223,9 @@ export default {
     `),
 
     [TYPE.externalAssertionLibraryError]: err => markup(err, `
-        ${escapeHtml(err.errMsg)} ${err.getCallsiteMarkup({ stackOnly: true })}
+        ${escapeHtml(err.errMsg)}
+
+        ${err.getCallsiteMarkup()}
```
Comment (inikulin) [inline src/errors/test-run/templates.js]: BTW, seems like we don't need `stackOnly` option anymore (need to check this), so we can remove it now from `getCallsiteMarkup`

**Comment by AlexanderMoskovkin** [comment]
FPR

**Comment by AlexanderMoskovkin** [comment]
Just an unstable client test

**Comment by inikulin** [comment]
lgtm

**Comment by VasilyStrelyaev** [comment]
lgtm

**Pull request merged**